Movistar Home ASKEY rev5 howto (#3)
zry98/movistar-home-hacks: RG3205W_rev5_howto.en.md

> # Make some use of Rev5
> **No need to open the box; no soldering; no root; no Linux**
> 
> *Just needs a bit of tapping, a (virtual) Bluetooth keyboard, lots of patience and spare time*
> 
> This is a quick howto documenting my experiments with an ASKEY (RG3205W) rev5 (Fab date: 2101).
> This wouldn't be possible without all the existing effort documented by the community, and their active support on the Telgram channel!
> 
> In fact, the below steps should also work with the Mitrastar, or the ASKEY rev4 versions; I just havent' tested with those units.
> While this setup has several limitations, one positive bit is that audio and camera work, something you currently lose if you install Linux on the Mitrastar.

Reviewer: ```suggestion
# Hacks for RG3205W Rev5

_Author:_ @alfredopironti

**No need to open the box; no soldering; no root; no Linux**

*Just needs a bit of tapping, a Bluetooth (or virtual) keyboard, lots of patience and spare time*

This is a quick how-to documenting my experiments with an ASKEY (RG3205W) rev5 (Fab date: 2101).
This wouldn't be possible without all the existing effort documented by the community, and their active support on the Telegram channel!

The hacks below should also work with the *MitraStar* IGW-5000 model with an Intel CPU, or the *ASKEY* RG3205W Rev4 version; I just havent' tested with those units.
While this setup has several limitations, no low-level components were modified, so the speaker, microphone and camera all work perfectly, something you currently lose if you install Linux on the IGW-5000.
```

## Escaping the Movistar Home kiosk
These initial steps are documented elsewhere, I'm reporing them here to make this howto self-contained.

### Connect a Bluetooth (virtual) keyboard to launch the embedded email client
Start with a vanilla Movistar Home (maybe do a factory reset to ensure you're starting from a known state).
As documented elsewhere in the docs, at this stage (and only at this stage), you'll need a Wi-Fi connection via the Movistar ISP,
so that the Movistar Home device can complete its initial setup process. Connect through the Movistar-provided Wi-Fi, and go through the initial setup.
Make sure the setup completes and you are in the Movistar Home homepage.

From there, go to the config and try to add a bluetooth speaker. When the Movistar Home is accepting Bluetooth connections, use a bluetooth keyboard to connect.
You should be able to use a physical keyboard. If you have a computer with MacOS, you can use the *keyPad* app;
it's handy as it also allows you to paste long strings (e.g. Bearer tokens).

> Once the Bluetooth keyboard is connected, press `<meta>`-E (*`<meta>`* is the Win key on Windows keyboards, or the Command key on MacOS),

Reviewer: ```suggestion
Once the Bluetooth keyboard is connected, press <kbd>Meta</kdb> and <kdb>E</kbd> (<kbd>Meta</kdb> is the <kbd>Win</kdb> key on a typical keyboard for Windows, or the <kdb>⌘ Command</kbd> key on a keyboard for Macs),
```

to open the email app on the Movistar Home device.

### Setup email to receive and install an App Store .apk
Configure the email app to connect to a remote mailbox. In my experience, *gmx.com* mailboxes will work fine (you can create one dedicated to
your device); make sure to use the *IMAP* protocol, as the *POP* one doesn't correctly download attachments for me.
In my experience, Outlook and Yahoo mail won't work (I always get a username/password error, even when using app passwords);
Gmail doesn't allow to attach .apk files, which are fundamental in our next step.

Go to the web-based gmx.com interface, and create an email (either save as draft, or send to yourself), where you attach the .apk of an app store.

> Many people that worked with the Movistar Home device have recommended Aptoide, which worked fine with me. Other options are F-droid which, however,

Reviewer: ```suggestion
Many people that worked with the Movistar Home device have recommended [Aptoide](https://aptoide.en.aptoide.com/app), which worked fine with me. Other options are [F-Droid](https://f-droid.org/en/) which, however,
```

may have a smaller selection of free-software-only apps.

> Now, go back to the Movistar Home device email app. Refresh (either the inbox, or the draft folder), and open the email with the *aptoide.apk*
> app store file. Open the attachment. This will install the app store.

Reviewer: ```suggestion
Now, back to the email app on your Movistar Home device. Refresh (either the inbox, or the draft folder), and open the email with the app store APK file. Open the attachment. This will install the app store.
```

But, yes, you read well: Automate will *resume* automations, not *restart* them at boot. What does this means?
Suppose your device was idle, with the screensaver active, and the power goes down. When the power comes back up, your device reboots... and the
screensaver never kicks in. This is because with the screensaver active, the automation was waiting for the `DREAMING_STOPPED` intent. After reboot,
the device has no screensaver active, so unless you run and then exit from it manually, the `DREAMING_STOPPED` intent will never be sent, deadlocking your resumed automation.

Current dirty solution: create another automation that waits for the `BOOT_COMPLETED` intent, then launches the Screen Saver, and loops waiting again for the boot intent.
(Another solution could be: create an automation that, on boot, kills and re-start all other automations -- requires more nodes that eat against the Automate free tier.)
Yes, you see why I can't be proud. But, again, it works, within the limitations of the OS running on the Movistar Home.

If you find any way of programmatically detecting when the device is idle, please let the community know!

> 

Reviewer: Is it possible to export & import the complete workflows from Automate as file? So others can replicate them easier.

Author: I'm working on it. I could dump individual flows, but need to check they restore properly. I also need to sanitize them (e.g. set a dummy location for the sunrise/sunset calculation).

Anyway, I do intend to share them. We could update this page once they're ready?

Author: Alright, just added the three flows I'm using (see 3rd commit at the bottom of this PR). If you're happy to merge now, I'll then prepare a further PR to update the text to link to the files and explain how to import and configure them in Automate, to make it more understandable.
